myia/public_api.py

```diff
@@ -603,7 +613,8 @@ def reshape(x, *shp):
 def scatter(self, dim, index, src):
     """Map of 'scatter' pytorch method."""
     if not P.hastype(src, APT):
-        src = P.scalar_to_array(P.scalar_cast(src, self.dtype), APT)
+        if P.hastype(self, APT) or P.hastype(src, AS):
+            src = P.scalar_to_array(P.scalar_cast(src, self.dtype), APT)
```

Comment:
Same idea as above.

```suggestion
        src = P.scalar_to_array(P.scalar_cast(src, self.dtype), operations.typeof(self))
```

Reply:
Proposed change doesn't seem to work. It throws this error:
```/Users/ethancaballero/mila-iqia/myia_outer/myia/myia/public_api.py:614
in scatter(
    self :: Float[32] x 3 x 4,
    dim :: <PendingFromList finished result=Int[64]> = 0,
    index :: Int[64] x 2 x 4,
    src :: Float[32] x 2 x 4
)
614: src = P.scalar_to_array(P.scalar_cast(src, self.dtype), operations.typeof(self))
================================================================================
in scalar_cast(Float[32] x 2 x 4, Ty(AbstractScalar(Float[32])))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
MyiaTypeError: Expected Number, not PyTorchTensor
```

Reply:
Which test?